docs/storage-surface.rst

```diff
@@ -0,0 +1,916 @@
+``gcloud.storage`` API
+======================
+
+Connection / Authorization
+--------------------------
+
+- Inferred defaults used to create connection if none configured explicitly
+
+  - credentials (derived from GAE / GCE environ if present).
+
+  - ``project`` (derived from GAE / GCE environ if present).
+
+  - ``bucket`` (derived from GAE / GCE environ if present).
+
+- By calling methods which require authentication
+
+  .. code-block:: python
+
+   >>> from gcloud import storage
+   >>> bucket = storage.get_bucket('my-bucket')
+
+  the default connection "just works" out of the box and will be lazily
+  loaded and stored after the first use.
+
+- Set defaults in a declarative fashion
+
+  .. code-block:: python
+
+   >>> storage.set_defaults()
+
+  or instead of using implicit behavior, pass in defaults
+
+  .. code-block:: python
+
+   >>> storage.set_defaults(connection=connection, project='some-project')
+   ...                      bucket=some_bucket)
+
+  though not all are needed
+
+  .. code-block:: python
+
+   >>> storage.set_defaults(project='some-project', bucket=some_bucket)
+
+- Declaratively set defaults one-by-one
+
+  .. code-block:: python
+
+   >>> storage.set_default_bucket(some_bucket)
+   >>> storage.set_default_connection(connection)
+   >>> storage.set_default_project(project)
+
+
+Manage buckets
+--------------
+
+Create a new bucket
+
+.. code-block:: python
+
+   >>> from gcloud import storage
+   >>> new_bucket = storage.create_bucket(bucket_name)
+
+.. warning::
+  If the bucket already exists, this method will throw an exception
+  corresponding to the `409 conflict`_ status code in the response.
+
+If you desire to be declarative, you may pass in a connection and a project
+to override the defaults
+
+.. code-block:: python
+
+   >>> new_bucket = storage.create_bucket(bucket_name, connection=connection,
+   ...                                    project=project)
+
+.. note::
+  All methods in the ``storage`` package accept ``connection`` as an optional
+  parameter. (Only ``create_bucket`` and ``list_buckets`` accept ``project``
```

Comment:
Hm - why do we differ on these? Is this on purpose or just something we eventually need to clean up ?